python/sbp/client/loggers/json_logger.py

```diff
@@ -62,13 +67,17 @@ def next(self):
 
     """
     for line in self.handle:
-      data = json.loads(line)
-      delta = data['delta']
-      timestamp = data['timestamp']
-      item = SBP.from_json_dict(data['data'])
       try:
-        yield (delta, timestamp, self.dispatcher(item))
-      except KeyError:
-        yield (delta, timestamp, item)
+        data = json.loads(line)
+        delta = data['delta']
+        timestamp = data['timestamp']
+        item = SBP.from_json_dict(data['data'])
+        try:
+          yield (delta, timestamp, self.dispatcher(item))
+        except KeyError:
+          yield (delta, timestamp, item)
+      except ValueError:
```

Comment:
I think we want to issue the warn and move on to the next line. We're doing line-based processing, so it's feasible that one-line could go bad without impacting the other lines (for whatever reason) - we should be robust to that. If we get a `ValueError` here, we have nothing to yield back to iterator - there are no values we can draw from the line. I don't think we need to differentiate here between the errors unless further experience suggests otherwise - bad JSON value or incomplete line, the line output should give us enough information. My thoughts anyway.